server/utilities/transaction.ts

```diff
@@ -0,0 +1,75 @@
+import { StatusCodes } from "http-status-codes";
+import { createCustomError } from "../../core";
+import { getTxById } from "../../src/db/transactions/getTxById";
+import { updateTx } from "../../src/db/transactions/updateTx";
+import { TransactionStatusEnum } from "../schemas/transaction";
+
+export const cancelTransactionAndUpdate = async (
+  queueId: string,
+  walletAddress: string,
+) => {
+  const txData = await getTxById({ queueId });
+
+  console.log("txData", txData);
+
+  if (!txData) {
+    const error = createCustomError(
+      `Transaction not found with queueId ${queueId}`,
+      StatusCodes.NOT_FOUND,
+      "QueueIdNotFound",
+    );
+    throw error;
+  }
+
+  let error = null;
+
+  switch (txData.status) {
+    case TransactionStatusEnum.Cancelled:
+      error = createCustomError(
+        `Transaction already cancelled with queueId ${queueId}`,
+        StatusCodes.BAD_REQUEST,
+        "TransactionAlreadyCancelled",
+      );
+      break;
+    case TransactionStatusEnum.Queued:
+      console.log("inside Mined/Queued");
```

Comment:
Ik this branch is WIP - just another reminder to remove these `console.log` before we ship

Reply:
yes yes. will remove debug & console.log(). as debug will start printing if someone changes the NODE_ENV on their docker image